packages/modules/Extension-Ads/src/contentScript/content.js

```diff
@@ -858,12 +890,27 @@ if ( typeof vAPI === 'object' && !vAPI.contentScript ) {
       debugLog.content && console.log(`extractContent: ${frameId}`,elt?elt:"",content,document)
       return content;
     },
-
-    getAdsCoordinates(element) {
+    async getAdsCoordinates(element) {
       // for iframe elements
       if (typeof element.getBoundingClientRect !== 'function') {
-        return null;
-      }
+        const src = window.location.href;
+
+        window.top.postMessage({
+            action: 'getCoordinates',
+            src: src
+        }, '*');
+
+        return new Promise((resolve) => {
```

Comment:
Create a separate method for this and move window.top.postMessage inside Promise. Name suggestion: getIframeElementCoordinates